plugins/SEO-Tool/js/main.js

```diff
@@ -0,0 +1,191 @@
+		Object.keys(siblingTabs).forEach((key, i) => {
+			siblingTabs[key].classList.remove('is-active');
+		});
+
+		elem.classList.add('is-active');
+		if(isElement(active) && !active.contains(elem))
+		{
+			active.classList.add('no-show');
+			active.style.display = 'none';
+			active = {};
+		}
+
+		if(subtabs === null)
+		{
+			if(!(elem.getAttribute('data-url') in reports))
+			{
+				document.getElementById('loader').style.display = 'block';
+				clear();
+			}
+
+			else
+			{
+				document.getElementById('loader').style.display = 'none';
+				populate(elem.getAttribute('data-url'), reports[elem.getAttribute('data-url')]);
+			}
+		}
+
+		else
+		{
+			subtabs.classList.remove('no-show');
+			subtabs.style.display = 'block';
+			active = subtabs;
+
+			const currTabs = document.getElementsByClassName('is-active');
+			Object.keys(currTabs).forEach((key, ind) => {
+				if(subtabs.contains(currTabs[key]))
+				{
+					changeActive(currTabs[key]);
+				}
+			});
+		}
+	};
+
+	function populate(link, report) {
+		lighthousePopulate(link, report['lighthouse']);
+		gscPopulate(link, report['gsc']);
+	};
+
+	const storage = window.localStorage, tabs = document.getElementsByClassName('v-tabs'), colors = ['red', 'orange', 'green'];
+	let active = {}, luColors = {};
+	const [pages, LUs] = parse(tabs);
+
+	const subArrs = splitToChunks([...pages], 5);
+	let reports = {};
+
+	function reportGen() {
+		const promises = subArrs.map(async (pages, i) => {
+			for(let i = 0; i < pages.length; i += 1)
+			{
+				const report = JSON.parse(storage.getItem(pages[i]));
+
+				if(expiryCheck(storage))
+				{
+					break;
+				}
+
+				if(report !== null && Object.keys(report.gsc).length && Object.keys(report.lighthouse).length)
+				{
+					reports[pages[i]] = {...report};
+				}
+
+				else
+				{
+					const lighthouseRes = await lighthouseApi(pages[i], commonData.apiKeys['lighthouse']), gscRes = await gscApi(pages[i], commonData.apiKeys['gsc']);
+					reports[pages[i]] = {
+						lighthouse: {...lighthouseRes},
+						gsc: {...gscRes}
+					};
+
+					storage.setItem(pages[i], JSON.stringify(reports[pages[i]]));
+				}
+
+				const mobPerfScore = reports[pages[i]]['lighthouse']['mobile']['Scores']['performance'], tab = document.querySelector(`[data-url='${pages[i]}']`), currColor = colorScheme(mobPerfScore);
+				let parentLU = null;
+
+				LUs.forEach((lu, ix) => {
+					const luElem = document.getElementById(lu + 'SubTabs');
+					if(luElem.contains(tab))
+					{
+						parentLU = document.querySelector(`[data-url='${lu}']`);
+						if(!(lu in luColors))
+						{
+							luColors[lu] = currColor;
+							parentLU.children[0].children[0].classList.add(colors[currColor]);
+						}
+
+						else if(luColors[lu] > currColor)
+						{
+							parentLU.children[0].children[0].classList.remove(colors[luColors[lu]]);
+							luColors[lu] = currColor;
+							parentLU.children[0].children[0].classList.add(colors[currColor]);
```

Comment:
Adding and removing classes has been done a few times on ''parentLU.children[0].children[0]". Can we take it out into a separate function with meaningful name so whatever we are trying to do is clear?